### PR TITLE
Chore/disk resize modal

### PR DESCRIPTION
**Pull request by saltcod**
Adds more detail to the disk size panels to help with confusion around disk auto-resizing.
Also connects the settings/database panel to the reports/database panel and allows customers to increase disk size from both places. 

1. changes the layout
2. adds Space used and Total size 
3. adds Space used and Total size to the database report
4. adds the Increase Disk Size modal to the db report so customers can upgrade there too
5. moved the "Importing a lot of data" Alert into this panel for better visiblity. I think it's getting lost when it's in a separate panel below, and is pretty critical info.
6. adds link to the [vacuum docs](https://supabase.com/docs/guides/platform/database-size#vacuum-operations) 

screenshot-2024-07-31-at-10 40 07

### CODE REVIEW

File: apps/studio/components/interfaces/Settings/Database/DiskSizeConfiguration.tsx [hunk 1]
```diff
@@ -86,66 +81,98 @@ const DiskSizeConfiguration = ({ disabled = false }: DiskSizeConfigurationProps)
...
     <div id="diskManagement">
       <FormHeader title="Disk management" />
       {projectSubscriptionData?.usage_billing_enabled === true ? (
         <div className="flex flex-col gap-3">
           <Panel className="!m-0">
             <Panel.Content>
-              <div className="grid grid-cols-1 items-center lg:grid-cols-3">
-                <div className="col-span-2 space-y-1">
+              <div className=" ">
+                <div className="">
```
Comment (alaister) [inline apps/studio/components/interfaces/Settings/Database/DiskSizeConfiguration.tsx:98]: Not sure why but I find this upsetting
```suggestion
              <div>
                <div>
```

File: apps/studio/components/interfaces/Settings/Database/DiskSizeConfiguration.tsx [hunk 2]
```diff
@@ -86,66 +81,98 @@ const DiskSizeConfiguration = ({ disabled = false }: DiskSizeConfigurationProps)
...
+
+                    <div className="col-span-8">
+                      <Alert_Shadcn_>
+                        <Info size={16} />
+                        <AlertTitle_Shadcn_>Importing a lot of data?</AlertTitle_Shadcn_>
+                        <AlertDescription_Shadcn_>
+                          <Markdown
+                            className="max-w-full"
+                            content={`
 We auto-scale your disk as you need more storage, but can only do this every 6 hours.
```
Comment (alaister) [inline apps/studio/components/interfaces/Settings/Database/DiskSizeConfiguration.tsx:160]: wdyt?
```suggestion
We auto-scale your disk as you need more storage, but can only do this once every 6 hours.
```

File: apps/studio/components/interfaces/Settings/Database/DiskSizeConfigurationModal.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+  const confirmResetDbPass = async (values: { [prop: string]: any }) => {
+    if (!projectRef) return console.error('Project ref is required')
+    const volumeSize = values['new-disk-size']
+    updateProjectUsage({ projectRef, volumeSize })
+  }
+
+  const currentDiskSize = project?.volumeSizeGb ?? 0
+  // to do, update with max_disk_volume_size_gb
+  const maxDiskSize = 200
```
Comment (alaister) [inline apps/studio/components/interfaces/Settings/Database/DiskSizeConfigurationModal.tsx:64]: Is this a needed todo before merging?

**Review by alaister — APPROVED** [review]
LGTM!

- Tested by pushing all the buttons on the disk resize panel and modal

also left a few nits:

File: apps/studio/components/interfaces/Settings/Database/DiskSizeConfiguration.tsx (hunk 1 above)
Comment (saltcod) [reply apps/studio/components/interfaces/Settings/Database/DiskSizeConfiguration.tsx:98]: hah! omg that's the worst! sorry! totally upsetting.

**Pull request merged**